lib/common/cpu.h

```diff
@@ -34,8 +34,8 @@ MEM_STATIC ZSTD_cpuid_t ZSTD_cpuid(void) {
     U32 f1d = 0;
     U32 f7b = 0;
     U32 f7c = 0;
-#if defined(_MSC_VER) && (defined(_M_X64) || defined(_M_IX86))
-#if !defined(__clang__) || __clang_major__ >= 16
+#if defined(_MSC_VER) && (defined(_M_X64) || defined(_M_IX86))                 \
+    && (!defined(__clang__) || __clang_major__ >= 16)
```

Comment:
As-is this leaves the CPUID for MSVC 32-bit builds unset. Instead, based on https://reviews.llvm.org/D101338, it looks like the clang bug is only present in 64-bit builds. So we could do

```
#if defined(_MSC_VER) && (defined(_M_X64) || defined(_M_IX86))
#if !defined(_M_X64) || !defined(__clang__) || __clang_major__ >= 16
...
#else
...
#endif
```

Reply:
Ok, I will change it